cli/flox-rust-sdk/src/models/environment/managed_environment.rs

```diff
@@ -465,42 +467,25 @@ impl ManagedEnvironment {
             .split('.')
             .next_back()
             .map(|hash| {
-                let links_dir = reverse_links_dir(flox);
-                let link = links_dir.join(hash);
-                std::fs::read_link(&link)
-                    .map_err(|e| ManagedEnvironmentError::ProjectNotFound { path: link, err: e })
+                let reg_path = env_registry_path(flox);
+                let reg: EnvRegistry = read_environment_registry(reg_path)?
+                    .ok_or(EnvRegistryError::EnvNotRegistered)?;
+                let hash = match hash.len() {
+                    // Older branches probably still have the longer hashes
+                    n if n < N_HASH_CHARS => Err(ManagedEnvironmentError::BadBranchName(
+                        String::from(branch_name),
+                    )),
+                    n if n > N_HASH_CHARS => Ok(hash.split_at(N_HASH_CHARS).0),
+                    n => Ok(hash),
+                }?;
```

Comment:
@mkenigs asked that we truncate the hashes so that they take up less (visual) space when included in `PATH` or when a user sees a path containing the hash.